boring/src/ssl/mod.rs

```diff
@@ -718,15 +718,15 @@ impl SslCurve {
     pub const X25519_KYBER768_DRAFT00: SslCurve =
         SslCurve(ffi::SSL_CURVE_X25519_KYBER768_DRAFT00 as _);
 
-    #[cfg(feature = "pq-experimental")]
+    #[cfg(all(not(feature = "fips"), feature = "pq-experimental"))]
```

Comment:
`not(feature = "fips")` prevents us from building this code whenever `"fips"` is enabled. This is necessary because this API is too new for older boringSSL builds.

We could weaken this constraint to `not(feature = "fips-compat")`, but this is somewhat of a broader change than what this PR is trying to do.

Note: this is not a breaking change because `"fips"` and `"pq-experimental"` were previously mutually exclusive.